schemas/private/ekscluster-kfd-v1alpha2.json

```
@@ -1192,10 +1206,18 @@
             }
           }
         },
+        "tsdbStartDate": {
+          "type": "string",
+          "format": "date",
+          "description": "The date loki have to switch to TSDB and schema v13"
```

Please remember that this field will end up:
- in the autocomplete of your code editor that pulls the schema to validate the file structure and contents.
- In KFDs documentation site, for example here: https://docs.kubernetesfury.com/docs/furyctl/providers/onpremises#specdistributionmodulesloggingloki

So, let's try to be a little more expressive and provide useful information to the user that has to fill this field. Right now is kind of the same as the key but with more words. I suggest something like:

```suggestion
          "description": "Starting from versions 1.28.4, 1.29.5 and 1.30.0 of KFD, Loki will change the time series database from BoltDB to TSDB and the schema from v11 to v13 that it uses to store the logs.\n\nThe value of this field will determine the date when Loki will start writing using the new TSDB and the schema v13, always at midnight UTC. The old TSDB and schema will be kept until they expire for reading purposes.\n\nValue must be a string in `ISO 8601` date format (`yyyy-mm-dd`). Example: `2024-11-18`."
```

Feel free to change anything, or all of the suggestion.

---

I think that we can introduce that, since the change if done incorrectly may result in loss of logs data, it will be very useful to have this in the autocomplete and on the KFD docs site. Thank you for pointing it out!

---

shipped 📦